mod/state-transition/pkg/core/state_processor_staking.go

```diff
@@ -38,26 +39,37 @@ func (sp *StateProcessor[
 	st BeaconStateT,
 	blk BeaconBlockT,
 ) error {
-	// Verify that outstanding deposits are processed up to the maximum number
-	// of deposits.
-	deposits := blk.GetBody().GetDeposits()
-	index, err := st.GetEth1DepositIndex()
+	// Verify that outstanding deposits matches those listed by contract
+	depositIndex, err := st.GetEth1DepositIndex()
 	if err != nil {
 		return err
 	}
-	eth1Data, err := st.GetEth1Data()
+	depositIndex++
```

Comment:
@calbera while adding unit tests spot this bug.
`st.GetEth1DepositIndex()` store the latest processed deposit, hence we need to move it up to process next.
This is similar to what we do in genesis